lib/src/nutripatrol_api_client.dart

```diff
@@ -0,0 +1,177 @@
+      path: '/api/v1/tickets/$ticketId',
+      uriHelper: uriHelper,
+    );
+    final Response response = await HttpHelper().doGetRequest(
+      uri,
+      uriHelper: uriHelper,
+    );
+    if (response.statusCode == 200) {
+      try {
+        final dynamic decodedResponse = HttpHelper().jsonDecodeUtf8(response);
+        return MaybeError<Ticket>.value(Ticket.fromJson(decodedResponse));
+      } catch (_) {
+        //
+      }
+    }
+    return MaybeError<Ticket>.responseError(response);
+  }
+
+  /// Get all tickets.
+  static Future<MaybeError<Tickets>> getTickets({
+    final NutripatrolTicketStatus status = NutripatrolTicketStatus.open,
+    final NutripatrolType type = NutripatrolType.image,
+    final int? page,
+    final int? pageSize,
+    final UriProductHelper uriHelper = uriHelperFoodProd,
+  }) async {
+    final Map<String, String> queryParameters = <String, String>{};
+    queryParameters['status'] = status.toString().split('.').last;
+    queryParameters['type_'] = type.toString().split('.').last;
+    if (page != null) queryParameters['page'] = page.toString();
+    if (pageSize != null) queryParameters['page_size'] = pageSize.toString();
+
+    final Uri uri = getUri(
+      path: '/api/v1/tickets',
+      queryParameters: queryParameters,
+      uriHelper: uriHelper,
+    );
+    final Response response = await HttpHelper().doGetRequest(
+      uri,
+      uriHelper: uriHelper,
+    );
+    if (response.statusCode == 200) {
+      try {
+        final dynamic decodedResponse = HttpHelper().jsonDecodeUtf8(response);
+        final List<dynamic> tickets = decodedResponse['tickets'];
+        return MaybeError<Tickets>.value(
+          Tickets.fromJson(<String, dynamic>{
+            'tickets': tickets,
+            'max_page': decodedResponse['max_page'],
+          }),
+        );
+      } catch (_) {
+        //
+      }
+    }
+    return MaybeError<Tickets>.responseError(response);
+  }
+
+  /// Create a Flag for a product.
+  ///
+  /// [flag] is the flag to create.
+  static Future<MaybeError<CreateFlag>> createFlag({
+    required final CreateFlagRequest flag,
+    final UriProductHelper uriHelper = uriHelperFoodProd,
+  }) async {
+    final Uri uri = getUri(
+      path: '/api/v1/flags',
+      uriHelper: uriHelper,
+    );
+    final Map<String, dynamic> body = <String, dynamic>{
+      if (flag.barcode != null) 'barcode': flag.barcode,
+      if (flag.comment != null) 'comment': flag.comment,
+      if (flag.confidence != null) 'confidence': flag.confidence,
+      if (flag.imageId != null) 'image_id': flag.imageId,
+      if (flag.reason != null) 'reason': flag.reason,
+      'type': flag.type,
+      'url': flag.url,
+      'user_id': flag.userId,
+      'device_id': flag.deviceId,
+      'source': flag.source,
+      'flavor': flag.flavor,
+    };
+    final Response response = await HttpHelper().doPostJsonRequest(
+        uri, jsonEncode(body),
+        uriHelper: uriHelper, bearerToken: "");
```

Comment:
`bearerToken: ""`? I don't think so.
Please consider using `doPostRequest`.
If you really need `doPostJsonRequest`, you'll have to do something cleaner, like an explicit `String? bearerToken`.